forte/data/data_store.py

```diff
@@ -0,0 +1,239 @@
+            attr_name (str): name of the attribute.
+            attr_value: value of the attribute.
+
+        """
+        # We retrieve the entry list from entry_dict using tid. We get its
+        # entry type. We then locate the index of the attribute using the entry
+        # type, field dictionary and attr_name, and update the attribute.
+
+        raise NotImplementedError
+
+    def get_attr(self, tid: int, attr_name: str):
+        r"""This function locates the entry list with tid and gets the value
+        of `attr_name` of this entry.
+
+        Args:
+            tid (int): Unique id of the entry.
+            attr_name (str): name of the attribute.
+
+        Returns:
+            The value of attr_name for the entry with tid.
+        """
+        # We retrieve the entry list from entry_dict using tid. We get its
+        # entry type. We then locate the index of the attribute using the entry
+        # type, field dictionary and attr_name, and get the attribute.
+
+        raise NotImplementedError
+
+    def delete_entry(self, tid: int):
+        r"""This function locates the entry list with tid and removes it
+        from the data store. It removes the entry list from both entry_dict
+        and the sortedlist of its type.
+
+        Args:
+            tid (int): Unique id of the entry.
+
+        """
+        # We retrieve the entry list from entry_dict using tid. We get its
+        # entry type, type id, begin and end indexes. Then, we find the
+        # `entry_type` sortedlist using type id. We bisect the sortedlist
+        # to find the entry list. We then remove the entry list from both
+        # entry_dict and the `entry_type` sortedlist.
+
+        raise NotImplementedError
+
+    def get_entry(self, tid: int) -> List:
+        r"""Look up the entry_dict with key `tid`.
+
+        Args:
+            tid (int): Unique id of the entry.
+
+        Returns:
+            The entry which tid corresponds to.
+
+        """
+        raise NotImplementedError
+
+    def get(
+        self, entry_type: Union[str, Type[EntryType]], **kwargs
+    ) -> Iterator[List]:
+        """
+        Implementation of this method should provide to obtain the entries of
+        type entry_type.
+
+        Args:
+            entry_type: The type of the entry to obtain.
+
+        Returns:
+            An iterator of the entries matching the provided arguments.
+
+        """
+        # We find the type id according to entry_type and locate sortedlist.
+        # We create an iterator to generate entries from the sortedlist.
+
+        raise NotImplementedError
+
+    def next_entry(self, tid: int) -> List:
+        r"""Get the next entry of the same type as the tid entry.
+
+        Args:
+            tid (int): Unique id of the entry.
+
+        Returns:
+            The next entry of the same type as the tid entry.
+
+        """
+        # We retrieve the entry list from entry_dict using tid. We get its
```

Comment:
The `next_entry` and `prev_entry` APIs should be designed with efficient access. When we use the `next_entry` API, we should assume we have the information of this entry, especially, its exact location in the data store. So we shouldn't redo biset here, but use the location information to directly get next.

Example usage of next_entry:
1. next_entry(exact_locatoin_value)
2. Sentence.next() # The class version of the Sentence can have a `next` function that directly calls the next_entry, note that no bisect should be needede.